partition_single_cell_mask produces rare error

In rare occasions partition_single_cell_mask produces an error

assert len(masks) == num_partitions, "Number of partitions(%s) does not match the number of masks (%s)." % (
AssertionError: Number of partitions(4) does not match the number of masks (1)

--> find good test case
--> print warning instead of error

--- a/.github/workflows/ci-unittest.yml
+++ b/.github/workflows/ci-unittest.yml
@@ -60,14 +60,14 @@ jobs:
     runs-on: macos-latest
     strategy:
       matrix:
-        python-version: ["3.8"]
+        python-version: ["3.8.18"]
 
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python 3.80
+      - name: Set up Python 3.8
         uses: actions/setup-python@v3
         with:
-          python-version: "3.8"
+          python-version: "3.8.18"
       - name: Install_macos
         run: |
           python -m pip install --upgrade pip

--- a/.github/workflows/ci-unittest.yml
+++ b/.github/workflows/ci-unittest.yml
@@ -57,17 +57,17 @@ jobs:
           python -m unittest discover src/polarityjam/test/
 
   build_macos:
-    runs-on: macos-latest
+    runs-on: macos-12
     strategy:
       matrix:
-        python-version: ["3.8.18"]
+        python-version: ["3.8"]
 
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python 3.8
+      - name: Set up Python 3.80
         uses: actions/setup-python@v3
         with:
-          python-version: "3.8.18"
+          python-version: "3.8"
       - name: Install_macos
         run: |
           python -m pip install --upgrade pip

--- a/src/polarityjam/compute/shape.py
+++ b/src/polarityjam/compute/shape.py
@@ -120,8 +120,11 @@ def partition_single_cell_mask(
     b = [pg_cent_a + int(major_axes_length), pg_cent_b]  # lies horizontally
     div_line = LineString([a, b])
 
+    # turn this line based on the cue direction
+    div_line = rotate(div_line, cue_direction, origin=(pg_cent_a, pg_cent_b))
+
     # determine number of divisions
-    divisors, div_angle = get_divisor_lines(a, cue_direction, div_line, num_partitions)
+    divisors, div_angle = get_divisor_lines(a, div_line, num_partitions)
 
     div_coords = []
     for div in divisors:
@@ -176,16 +179,12 @@ def partition_single_cell_mask(
     return masks, polygons, contours
 
 
-def get_divisor_lines(
-    origin: List[int], cue_direction: int, div_line: LineString, num_partitions: int
-):
+def get_divisor_lines(origin: List[int], div_line: LineString, num_partitions: int):
     """Rotates a line from an origin point on the line based on a cue direction as often as specified.
 
     Args:
         origin:
             origin point on the line
-        cue_direction:
-            The orientation of the cue (e.g. flow)
         div_line:
             shapely line object
         num_partitions:
@@ -196,7 +195,7 @@ def get_divisor_lines(
 
     """
     div_angle = int(360 / num_partitions)
-    cumulative_angle = cue_direction + int(div_angle / 2)
+    cumulative_angle = int(div_angle / 2)
     divisors = []
     while cumulative_angle < 360:
         divisors.append(rotate(div_line, cumulative_angle, origin=origin))

--- a/src/polarityjam/controller/plotter.py
+++ b/src/polarityjam/controller/plotter.py
@@ -11,6 +11,7 @@ import numpy as np
 import pandas
 import scipy.ndimage as ndi
 from matplotlib import pyplot as plt
+from shapely.affinity import rotate
 from shapely.geometry import LineString
 
 from polarityjam.compute.shape import get_divisor_lines
@@ -2151,12 +2152,15 @@ class Plotter:
             b = [x0 + row["cell_major_axis_length"], y0]  # lies horizontally
             ground_line = LineString([a, b])
 
-            d_lines, _ = get_divisor_lines(a, params.cue_direction, ground_line, 4)
+            # rotate ground line based on cue_direction
+            ground_line = rotate(ground_line, params.cue_direction, origin=a)
+
+            d_lines, _ = get_divisor_lines(a, ground_line, 4)
             for d_line in d_lines:
                 x1, y1 = (i[0] for i in d_line.boundary.centroid.coords.xy)
                 ax[1].plot((x0, x1), (y0, y1), "--r", linewidth=0.5)
 
-            d_lines, _ = get_divisor_lines(a, params.cue_direction, ground_line, 2)
+            d_lines, _ = get_divisor_lines(a, ground_line, 2)
             for d_line in d_lines:
                 x1, y1 = (i[0] for i in d_line.boundary.centroid.coords.xy)
                 ax[0].plot((x0, x1), (y0, y1), "--r", linewidth=0.5)

